pkg/operator/reports.go

```diff
@@ -91,8 +91,26 @@ func (op *Reporting) syncReport(logger log.FieldLogger, key string) error {
 		}
 		return err
 	}
+
 	sr := report.DeepCopy()
 
+	// check if the report is expired
+	if report.DeletionTimestamp == nil {
+		if reportExpired := isReportExpired(logger, report, time.Now()); reportExpired {
+			if reportIsNotInput := isReportNotUsedAsInput(report, op); reportIsNotInput {
+				op.deleteReport(sr)
+				newDeletionTimestamp := metav1.Now()
+				report.SetDeletionTimestamp(&newDeletionTimestamp)
+				op.meteringClient.MeteringV1().Reports(report.Namespace).
+					Delete(context.TODO(), report.Name, metav1.DeleteOptions{})
```

Comment:
Hmm, this seems strange to me looking at the other `delete<resource>` methods we expose. I wonder if it makes sense to encapsulate this logic in another reporting method that's responsible for making the client delete call. Then, the `op.deleteReport` method call would call that method instead of `syncReport`.

Reply:
This PR doesn't touch `op.deleteReport` so far, but what I noticed in practice is that `op.deleteReport` only seems to do work against the queue, making it so report is no longer enqueued if you call it. But it left the report installed in the cluster. Are you saying we should move the metering client delete (that actually remove the Report from the cluster into `op.deleteReport` such that that method become comprehensive? I think it makes sense, less error prone for the function user, just clarifying.